File: GreenHouseController/greenhousecontroller/src/main/java/com/klnvch/greenhousecontroller/AppSettings.java
package com.klnvch.greenhousecontroller;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.annotation.NonNull;

public class AppSettings {
    private static final String SHARED_PREFERENCES_NAME = "settings";
    private static final String KEY_DEVICE_ADDRESS = "KEY_DEVICE_ADDRESS";

    private final SharedPreferences sharedPreferences;

    private AppSettings(@NonNull Context context) {
        sharedPreferences = context.getSharedPreferences(
                SHARED_PREFERENCES_NAME, Context.MODE_PRIVATE);
    }

    @NonNull
    public synchronized static AppSettings getInstance(@NonNull Context context) {
        return new AppSettings(context.getApplicationContext());
    }

    String getDeviceAddress() {
        return sharedPreferences.getString(KEY_DEVICE_ADDRESS, "98:D3:33:F5:A3:24");
    }

    void setDeviceAddress(String deviceAddress) {
        sharedPreferences.edit().putString(KEY_DEVICE_ADDRESS, deviceAddress).apply();
    }
}

File: GreenHouseController/greenhousecommon/src/main/java/com/klnvch/greenhousecommon/db/AppSettings.kt
package com.klnvch.greenhousecommon.db

import android.content.Context
import android.content.SharedPreferences

class AppSettings(context: Context) {
    var sharedPreferences: SharedPreferences =
        context.getSharedPreferences(NAME, Context.MODE_PRIVATE)

    fun getDeviceId(): String {
        return sharedPreferences.getString(KEY_DEVICE_ID, null) ?: DEFAULT_DEVICE_ID
    }

    fun setDeviceId(deviceId: String?) {
        sharedPreferences.edit().putString(KEY_DEVICE_ID, deviceId).apply();
    }

    fun getStartTime(): Long {
        return sharedPreferences.getLong(KEY_START_TIME, 0)
    }

    fun setStartTime(startTime: Long) {
        sharedPreferences.edit().putLong(KEY_START_TIME, startTime).apply()
    }

    fun getChartTimeInterval(): Long {
        return sharedPreferences.getLong(KEY_CHART_TIME_INTERVAL, WEEK)
    }

    fun setChartTimeInterval(timeInterval: Long) {
        sharedPreferences.edit().putLong(KEY_CHART_TIME_INTERVAL, timeInterval).apply()
    }

    companion object {
        private const val NAME = "settings"
        private const val KEY_DEVICE_ID = "deviceId"
        private const val DEFAULT_DEVICE_ID = "test"
        private const val KEY_START_TIME = "startTime";
        private const val KEY_CHART_TIME_INTERVAL = "chartTimeInterval";

        const val HOUR = (60 * 60 * 1000).toLong()
        const val DAY = 24 * HOUR
        const val WEEK = 7 * DAY
        const val MONTH = 30 * WEEK
    }
}
